apps/prairielearn/src/pages/course_instance/lookup/lookupByEnrollmentCode.ts

```diff
@@ -18,11 +22,32 @@ router.get(
     const { code, course_instance_id: courseInstanceId } = LookupCodeSchema.parse(req.query);
 
     // Look up the course instance by enrollment code
-    const courseInstance = await selectOptionalCourseInstanceByEnrollmentCode(code);
+    const courseInstance = await selectOptionalCourseInstanceByEnrollmentCode({
+      enrollment_code: code,
+      requestedRole: 'System',
+      authzData: dangerousFullSystemAuthz(),
+    });
+    if (!courseInstance) {
+      // User-facing terminology is to use "course" instead of "course instance"
+      res.status(404).json({
+        error: 'No course found with this enrollment code',
+      });
+      return;
+    }
 
-    // User-facing terminology is to use "course" instead of "course instance"
+    const { authzData } = await buildAuthzData({
+      authn_user: res.locals.authn_user,
+      course_id: null, // Inferred via course_instance_id
+      course_instance_id: courseInstanceId ?? null,
```

Comment:
Something feels off here. We sort of have two course instance IDs: the one from `req.query`, and an implicit one from the `courseInstance` that we read above. In the case where there is _not_ one in the query, we'll end up calling `buildAuthzData` with both a null `course_id` and a null `course_instance_id`, which will (hopefully) immediately blow up in our face.

On the last note there: it feels like either `buildAuthzData` or `selectAuthzData` should be explicitly checking for and handling the case of `course_id == null && course_instance_id == null`?